### ISSUE TITLE
`manual_div_ceil` should warn on manual implementations using `next_multiple_of`

### ISSUE THREAD

**Issue by mnemonikr** [rust-lang/rust-clippy#16219]
### Summary

This should be applicable for both unsigned and signed types (the latter with feature `int_roundings`):

```
#![feature(int_roundings)]
let a: i32 = 8;
let b = 3;

assert_eq!(a.div_ceil(b), a.next_multiple_of(b) / b);
assert_eq!(a.div_ceil(-b), a.next_multiple_of(-b) / -b);
assert_eq!((-a).div_ceil(b), (-a).next_multiple_of(b) / b);
assert_eq!((-a).div_ceil(-b), (-a).next_multiple_of(-b) / -b);
```

[Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2024&gist=5be9fb1fa42e29e97e6024604b2727cc)

Using `div_ceil` in this case has the advantage of avoiding the subtle issue that can occur if `a.next_multiple_of(b)` overflows.

### Lint Name

manual_div_ceil

### Reproducer

I tried this code:

```rust
let a: u32 = 8;
let b = 3;
let c = a.next_multiple_of(b) / b;
```

I expected to see this happen:

```
clippy: manually reimplementing `div_ceil`
...
clippy: consider using `.div_ceil()`: `a.div_ceil(b)` [manual_div_ceil]
```

Instead, this happened: The lint did not fire.

### Version

```text
rustc 1.92.0 (ded5c06cf 2025-12-08)
binary: rustc
commit-hash: ded5c06cf21d2b93bffd5d884aa6e96934ee4234
commit-date: 2025-12-08
host: x86_64-unknown-linux-gnu
release: 1.92.0
LLVM version: 21.1.3
```

**Comment by mnemonikr** [comment]
I can pick up implementing a fix for this unless there are any objections. Will post a PR shortly.
